fabric/Makefile

```diff
@@ -0,0 +1,48 @@
+# Copyright IBM Corp. All Rights Reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+TOP = ..
+include $(TOP)/build.mk
+
+FABRIC_VERSION ?= v2.0.1
+FABRIC_PEER_BIN = $(FABRIC_PATH)/build/bin/peer
+FABRIC_TLCC = $(FABRIC_PATH)/core/scc/tlcc
+
+FABRIC_PATCHED = $(FABRIC_PATH)/.patched
+
+build: peer native
+
+patch: $(FABRIC_PATCHED)
+
+$(FABRIC_PATCHED):
+	cd $(FABRIC_PATH) && \
+	git am $(FPC_PATH)/fabric/*.patch > $@
+	cat $@
+
+clean-patch:
+	cd $(FABRIC_PATH) && \
+	git reset --hard $(FABRIC_VERSION)
```

Comment:
I agree that this is a bit too aggressive. I propose the following:

- we checkout fabric v2.0.1 and create a branch called v2.0.1-fpc
- we only apply the patches on that branch
- we only apply the patches if HEAD points to tag v2.0.1
- we only perform cleaning if we are on that branch
- we remove clean-patch from clean target but keep it for local development

Reply:
a few thoughts:
- dropping clean-patch gives you trouble when you add patches and/or switch between different versions. Doing as now is robust (though dangerous).
- to make the clean-branch less dangerous, we could actually two slightly different things than above
  - instead of (or maybe in addition to) branching on v2.0.1 we could add a branch (or tag?) _after_ applying the patches and make sure that we do a clean-branch only if we are on that branch, e.g., test with something along the lines of, ` if [ "fpc-patched-branch" == "$(git rev-parse --abbrev-ref HEAD)" ]; then .... fi`
  - make sure we do not have any modifications which get overriden, e.g., along the lines of `if [ -z "$(git status -s | grep -v '\?')" ]; then....  fi`

Reply:
actually, better idea instead of add branch/tag after applying patches, is simply to add the commit id to the (currently empty) patched-file and then read that in the clean and compare with the current commit. (obviously you still want to do the no-modification test ...)

Reply:
I notice that you actually ask now interactively. This is robust but interaction during a makefile is not super elegant and might cause some potential trouble when make is run in scripts?

Reply:
sounds like this mechanism still needs work.

Reply:
I added the modification test and check that patch commits are available on that branch.

I suggest that we stop making this is more robust and create a separate issue/PR for this as it seems to be beyond the scope if this PR.

Reply:
I concur.

Reply:
hmm, not sure whether it is really beyond the scope given that this just part of the whole refactoring of the fabric build.  That said, I guess instead of going forth and back here, i guess we can table it and i will push a corresponding PR with my proposal once it is merged ...